src/include/lean/lean.h

```diff
@@ -398,6 +398,13 @@ static inline unsigned lean_ptr_other(lean_object * o) {
    small objects is a multiple of LEAN_OBJECT_SIZE_DELTA */
 LEAN_EXPORT size_t lean_object_byte_size(lean_object * o);
 
+/* Returns the size of the salient part of an object's storage,
+   i.e. the parts that contribute to the value representation;
+   padding or unused capacity is excluded. Operations that read
+   from an object's storage must only access these parts, since
+   the non-salient parts may not be initialized. */
+LEAN_EXPORT size_t lean_object_data_size(lean_object * o);
```

Comment:
Removing the `byte` part from the name somewhat suggests this is not measuring bytes. Can we find a better name here, perhaps `lean_object_data_byte_size`?

Reply:
We can absolutely revise the names! I thought it was nice for "data" and "byte" to have the same length, but clarity should come first. Do you think "data_byte_size" is good (esp. when contrasted with the existing "byte_size")?

It seems reasonable that the longer term is more qualified, i.e. only the "data bytes" are being considered.

Reply:
Yes, I'm happy with `lean_object_data_byte_size`

Reply:
Done!